test/createService.spec.js

```diff
@@ -0,0 +1,48 @@
+/* global describe, it */
+import chai, { expect } from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import createService from '../src/createService';
+chai.use(sinonChai);
+
+describe('createService', () => {
+  const mySpec = {
+    initialize: sinon.stub(),
+    customFunction1() { return 'value1'; },
+    customFunction2() { return 'value2'; },
+    returnsThis() { return this; }
+  };
+
+  const myOptions = {
+    app: {}
+  };
+
+  const MyService = createService(mySpec);
+  const myServiceInstance = new MyService(myOptions);
+
+  it('throws error when an app instance is not provided', () => {
+    expect(() => new MyService()).to.throw(Error, 'App instance not provided.');
+  });
+
+  it('creates instance from my service', () => {
+    expect(myServiceInstance).to.be.instanceOf(MyService);
+  });
+
+  it('executes the initialize() at construction', () => {
+    expect(mySpec.initialize).to.be.callCount(1);
```

Comment:
We had that before on our internal repo, but since we're in favour of functions instead of expressions ( you can see it based on our eslint rule `no-unused-expressions` ).

Also, we had some problems where the `called.once` was not validating properly ( sometimes the callCount was bigger than 1 ).

I would prefer to stay away from expressions and use functions since we're pretty much into functional programming/pure functions as well :)